check_test.go

```diff
@@ -0,0 +1,134 @@
+package subtest_test
+
+import (
+	"testing"
+
+	"github.com/searis/subtest"
+)
+
+func TestDeepEqual(t *testing.T) {
+	// Demonstrates GWT-style sub-tests.
+
+	t.Run("when checking bool(true)", func(t *testing.T) {
+		v := true
+		vf := subtest.Value(v)
+
+		t.Run("then it should match true", vf.DeepEqual(true))
+		t.Run("then it should match true", subtest.DeepEqual(true).Test(v)) // equivalent
```

Comment:
I think `subtest.Value(v).DeepEqual(true)` reads a lot better than `subtest.DeepEqual(true).Test(v)`. What's the reason for the second format? And what's the thinking for that specific way to construct the statement?

Reply:
I agree the first syntax is most clear. But it was good to experiment with both

The second syntax was just renamed to `TestValue(v)` or `Test(subtest.Value(v))`. The reasoning behind the second syntax is more conceptual than useful; it is _converting_ a `CheckFunc` instance to a _test function_ against a value. It is a method (or function) that we need, but it doesn't strictly need to be _public_.

I think the second syntax could _potentially_ be more clear in some cases. E.g. when checking many values against the same test, such as perhaps in classical Go table tests. 

I.e. I think this is slightly more readable:

```go
checkLen := checkBetween(10, 1000) // some custom CheckFunc instnace.
for name, tt := range tests {
   t.Run(name, checkLen.TestValue(len(tt.Bytes))
}
```
Than the slightly longer version we would have to do otherwise:

```go
checkLen := checkBetween(10, 1000) // some custom CheckFunc instnace.
for name, tt := range tests {
    t.Run(name, subtest.Value(len(tt.Bytes)).Test(checkLen)
}
```

Reply:
For now documented the latter syntax as "experimental" with warnings like "may be removed in the future". Sounds good?

After writing some tests, if we don't end up wanting to use it, we remove it.

Reply:
FYI, I have now remove the `Test` / `TestValue` methods on `CheckFunc`. 

The _two_ documented syntaxes for initializing _tests_ are now:
```go
t.Run(name, subtest.Value(v).DeepEqual(42)
```
And the longer syntax intended mainly for custom checks:
```go
t.Run(name, subtest.Value(v).Test(subtest.DeepEqual(42))
```

The latter syntax can also be useful with check "middleware" (not part of this PR, but got it working in a different repo):
```go
t.Run(name, subtest.Value(`{"foo": "bar"}`).Test(subjson.OnMap(subteset.Schema{
    Fields: subtest.Fields{
        "foo": subjson.OnString(subtest.DeepEqual("bar")),
    },
}.ValidateMap()))
```